src/streamable/de.rs

```diff
@@ -116,11 +116,11 @@ impl<'de, 'a> serde::Deserializer<'de> for &'a mut ChiaDeserializer<'de> {
         visitor.visit_i8(self.read_u8()? as i8)
     }
 
-    fn deserialize_unit<V>(self, visitor: V) -> Result<V::Value>
+    fn deserialize_unit<V>(self, _visitor: V) -> Result<V::Value>
     where
         V: serde::de::Visitor<'de>,
     {
-        visitor.visit_unit()
+        Err(Error::NotSupported)
```

Comment:
this was an inconsistency. The serializer doesn't support this, so I made the deserializer also fail (and the type blow as well)